### ISSUE TITLE
mAp in custom dataset

### ISSUE THREAD

**Issue by 100330706** [AlexeyAB/darknet#2560]
Hey!

So far I have trained a custom training set composed of 5 different classes. The weights were saved in the backup directory.

This is my obj.data:

`classes= 5
train  = data/train.txt
valid  = data/test.txt
names = data/obj.names
backup = backup/`

And in test.txt I have all the names of the test images.

When doing:

`./darknet detector valid ./data/obj.data ./cfg/yolo-obj.cfg ./backup/yolo-obj_20000.weights `

I get the results saved on the results directory. But how can I obtain the mAp or any way to visualize the results?

Regards

**Comment by AlexeyAB** [comment]
@100330706 Hi,

If you use this repository https://github.com/AlexeyAB/darknet
then just run this command:
`./darknet detector map ./data/obj.data ./cfg/yolo-obj.cfg ./backup/yolo-obj_20000.weights`

----

Also during training you can use `-map` flag and you will see mAP chart https://github.com/AlexeyAB/darknet#when-should-i-stop-training
`./darknet detector train ./data/obj.data ./cfg/yolo-obj.cfg -map`

![68747470733a2f2f6873746f2e6f72672f776562742f79642f766c2f61672f7964766c616775746f66327a636e6a6f64737467726f656e3861632e6a706567](https://user-images.githubusercontent.com/4096485/54025518-d3321a80-41ab-11e9-9dc6-ea9c234c27c6.jpg)

**Comment by arkjiang** [comment]
@AlexeyAB  is it possible that this mAP plot is saved when training on cloud ?

**Comment by AlexeyAB** [comment]
@arkjiang Yes, just compile Darknet with `OPENCV=1` in the Makefile, and run training command with flags `-map -dont_show`

**Comment by arkjiang** [comment]
@AlexeyAB im testing mjpeg server on cloud. 
It only loads top 20% of the image every time. Could it be due to each iteration finishes very quickly and the old mjpeg server is killed before the image is received in host ?
In addition, i'm trying to find the path where the loss/mAP image is saved when -map -dont_show flag is enabled.  
Thanks so much

**Comment by AlexeyAB** [comment]
@arkjiang 

* What commands do you use?

* If you compiled with OPENCV=1 in the Makefile, and **run training with flag `-map`** then `chart.png` (with points of loss/mAP) will be saved automatically for each 100 iterations.

Also you can run training command with ` | tee -a log.txt` so all training log will be saved to the log.txt file.
